src/collections/maple_tree/mod.rs

```diff
@@ -104,66 +131,307 @@ impl<K, V> SparseLeaf<K, V>
             }
         }
 
-        // If there is no slot, return a failure?
+        return None;
+    }
+
+    pub fn search(&mut self, k: &K) -> Option<&V> {
+        for i in 0..CAPACITY {
+            match &self.key[i] {
+                M::Some(v) => {
+                    if v == k {
+                        match &self.value[i] {
+                            M::Some(v) => {
+                                return Some(v);
+                            }
+                            M::None => {
+                                return None;
```

Comment:
If you have a return None here, on the first "None" you would return. Because of the design of this, the following is valid:

[Some(1), Some(2), None, Some(4)]

This can be easily seen by insert 1, 2, 3, 4, remove 3, search 4. So perhaps a test case for that sequence ....

Reply:
The code will only get to that return None on line 147 if the key has already been matched, line 141 makes sure the keys match so unless they are searching for a None then I don't think there will be a problem?
 I am happy to write the test case for that tho, and also to your point in the next comment, I guess the value shouldn't ever be None so for line 147 I guess a panic could be necessary.